src/engine/cachingreader/cachingreaderchunk.cpp

```diff
@@ -49,34 +48,51 @@ mixxx::IndexRange CachingReaderChunk::bufferSampleFrames(
         mixxx::SampleBuffer::WritableSlice tempOutputBuffer) {
     DEBUG_ASSERT(m_index != kInvalidChunkIndex);
     const auto sourceFrameIndexRange = frameIndexRange(pAudioSource);
-    mixxx::AudioSourceStereoProxy audioSourceProxy(
-            pAudioSource,
-            tempOutputBuffer);
-    DEBUG_ASSERT(
-            audioSourceProxy.getSignalInfo().getChannelCount() ==
-            kChannels);
-    m_bufferedSampleFrames =
-            audioSourceProxy.readSampleFrames(
-                    mixxx::WritableSampleFrames(
-                            sourceFrameIndexRange,
-                            mixxx::SampleBuffer::WritableSlice(m_sampleBuffer)));
+
+    if (pAudioSource->getSignalInfo().getChannelCount() %
+                    mixxx::audio::ChannelCount::stereo() !=
+            0) {
+        // This happens if the audio source only contain a mono channel, or an
+        // uneven number of channel
+        mixxx::AudioSourceStereoProxy audioSourceProxy(
+                pAudioSource,
+                tempOutputBuffer);
+        DEBUG_ASSERT(
+                audioSourceProxy.getSignalInfo().getChannelCount() ==
+                mixxx::audio::ChannelCount::stereo());
```

Comment:
> I am concerned that this will leads to out of bounds access later. But probably this for no reason? 

Sorry - which part? That code should be strictly equivalent to what it used to be.

> Where is the code that handles to throw away or zero out channels?

I have added a VERIFY assert to disqualify zero channel in the parent owning pAudioSource which hopefully should be enough. 
AFAIK, the expected behaviour of `AudioSource::readSampleFrames` is to return the range of available sample, which should be zero in this case, so adding further assert here feels a little late and redundant